src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java

```diff
@@ -0,0 +1,67 @@
+package edu.ucsb.cs156.team02.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import edu.ucsb.cs156.team02.ControllerTestCase;
+import edu.ucsb.cs156.team02.entities.CollegiateSubreddit;
+import edu.ucsb.cs156.team02.entities.User;
+import edu.ucsb.cs156.team02.repositories.CollegiateSubredditRepository;
+import edu.ucsb.cs156.team02.repositories.UserRepository;
+import edu.ucsb.cs156.team02.testconfig.TestConfig;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@WebMvcTest(controllers = CollegiateSubredditController.class)
+@Import(TestConfig.class)
+public class CollegiateSubredditControllerTests extends ControllerTestCase {
+    
+    @MockBean
+    CollegiateSubredditRepository collegiateSubredditRepository;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @WithMockUser(roles = { "USER" })
```

Comment:
Can you also do a test for each type for admin?

Reply:
I talked to Andrew and he gave me the impression that for this project, we don't have to deal with admins